Source/WebKit/UIProcess/WebProcessProxy.cpp

```diff
@@ -1367,11 +1373,15 @@ bool WebProcessProxy::isResponsive() const
     return responsivenessTimer().isResponsive() && m_backgroundResponsivenessTimer.isResponsive();
 }
 
-void WebProcessProxy::didDestroyUserGestureToken(uint64_t identifier)
+void WebProcessProxy::didDestroyUserGestureToken(PageIdentifier pageID, uint64_t identifier)
 {
     ASSERT(UserInitiatedActionMap::isValidKey(identifier));
+    auto authorizationTokenMapByPage = m_userInitiatedActionByAuthorizationTokenMap.get(pageID);
     if (auto removed = m_userInitiatedActionMap.take(identifier); removed && removed->authorizationToken())
-        m_userInitiatedActionByAuthorizationTokenMap.remove(*removed->authorizationToken());
+        authorizationTokenMapByPage.remove(*removed->authorizationToken());
```

Comment:
This seems like a bug, you're modifying a copy of the HashMap instead of the HashMap that's actually stored in the data member.